process/block/sovereignChainBlock_test.go

```diff
@@ -897,31 +897,35 @@ func TestSovereignShardProcessor_CreateBlock(t *testing.T) {
 func TestSovereignShardProcessor_ProcessBlock(t *testing.T) {
 	t.Parallel()
 
-	t.Run("nil header handler should error", func(t *testing.T) {
+	t.Run("nil header handler, should error", func(t *testing.T) {
 		sovArgs := createSovChainBlockProcessorArgs()
 		scbp, err := blproc.NewSovereignChainBlockProcessor(sovArgs)
 		require.Nil(t, err)
 
-		_, _, err = scbp.ProcessBlock(nil, &block.Body{}, haveTime)
+		var headerHandler data.HeaderHandler
+		_, _, err = scbp.ProcessBlock(headerHandler, &block.Body{}, haveTime)
+		require.Nil(t, headerHandler)
 		require.Equal(t, process.ErrNilBlockHeader, err)
```

Comment:
This is still wrong. You are declaring `headerHandler` nil variabile, then check it as being nil again.
What I meant by previous comment from here: https://github.com/multiversx/mx-chain-sovereign-go/pull/62#discussion_r2063366063

Is that instead of having

```go
_, _, err = scbp.ProcessBlock(headerHandler, &block.Body{}, haveTime)
```

You should have

```go
header, body, err = scbp.ProcessBlock(headerHandler, &block.Body{}, haveTime)
```